src/Groups/matrices/matrix_manipulation.jl

```diff
@@ -227,23 +227,29 @@ end
 Base.getindex(V::AbstractAlgebra.Generic.FreeModule, i::Int) = gen(V, i)
 
 # scalar product
-Base.:*(v::AbstractAlgebra.Generic.FreeModuleElem{T},u::AbstractAlgebra.Generic.FreeModuleElem{T}) where T <: FieldElem = (v.v*transpose(u.v))[1]
+Base.:*(v::AbstractAlgebra.Generic.FreeModuleElem{T},u::AbstractAlgebra.Generic.FreeModuleElem{T}) where T <: RingElem = (v.v*transpose(u.v))[1]
+#T do we want this at all? (type piracy?)
```

Comment:
It is indeed type piracy, see PR #760 

we should migrate this and more of the methods below into AA or Nemo (as appropriate)

Reply:
Concerning this particular method, I think we should just remove this method.
This is my understanding of @thofma's comment

> Vector * Vector is not defined and I don't think anyone suggested implementing it.

in the discussion of oscar-system/Oscar.jl/issues/872